qcodes/instrument_drivers/QDev/QDac_channels.py

```diff
@@ -0,0 +1,684 @@
+        """
+        Instantiates the instrument.
+
+        Args:
+            name (str): The instrument name used by qcodes
+            address (str): The VISA name of the resource
+            num_chans (int): Number of channels to assign. Default: 48
+            update_currents (bool): Whether to query all channels for their
+                current current value on startup. Default: True.
+
+        Returns:
+            QDac object
+        """
+        super().__init__(name, address)
+        handle = self.visa_handle
+
+        # This is the baud rate on power-up. It can be changed later but
+        # you must start out with this value.
+        handle.baud_rate = 480600
+        handle.parity = visa.constants.Parity(0)
+        handle.data_bits = 8
+        self.set_terminator('\n')
+        # TODO: do we want a method for write termination too?
+        handle.write_termination = '\n'
+        # TODO: do we need a query delay for robust operation?
+        self._write_response = ''
+
+        # The following bool is used in self.write
+        self.debugmode = False
+
+        if self._get_firmware_version() < 0.170202:
+            raise RuntimeError('''
+                               Obsolete QDAC Software version detected.
+                               QCoDeS only supports version 0.170202 or newer.
+                               Contact rikke.lutge@nbi.ku.dk for an update.
+                               ''')
+
+        self.num_chans = num_chans
+
+        # Assigned slopes. Entries will eventually be [chan, slope]
+        self._slopes = []
+        # Function generators (used in _set_voltage)
+        self._fgs = set(range(1, 9))
+        self._assigned_fgs = {}  # {chan: fg}
+        # Sync channels
+        self._syncoutputs = []  # Entries: [chan, syncchannel]
+
+        self.chan_range = range(1, 1 + self.num_chans)
+        self.channel_validator = vals.Ints(1, self.num_chans)
+
+        channels = ChannelList(self, "Channels", QDacChannel,
+                               snapshotable=False,
+                               paramclass=QDacMultiChannelParameter)
+
+        for i in self.chan_range:
+            channel = QDacChannel(self, 'chan{}'.format(i), i)
+            channels.append(channel)
+            # Should raise valueerror if name is invalid (silently fails now)
+            self.add_submodule('ch{:02}'.format(i), channel)
+        channels.lock()
+        self.add_submodule('channels', channels)
+
+        for board in range(6):
+            for sensor in range(3):
+                label = 'Board {}, Temperature {}'.format(board, sensor)
+                self.add_parameter(name='temp{}_{}'.format(board, sensor),
+                                   label=label,
+                                   unit='C',
+                                   get_cmd='tem {} {}'.format(board, sensor),
+                                   get_parser=self._num_verbose)
```

Comment:
They could be channelised, but noone uses them, so I think it's okay to leave them for now.